### ISSUE TITLE
pscoast +l or +r option on -A causes segmentation fault

### ISSUE THREAD

**Issue by liamtoney** [GenericMappingTools/gmt#231]
**Description of the problem**

I'm appending `+l` or `+r` to the `-A` option of `pscoast` to specify just regular lakes or just river-lakes (following the docs [here](http://gmt.soest.hawaii.edu/doc/5.4.4/pscoast.html#optional-arguments)), but I get a segmentation fault error message.

**Full script that generated the error**

```shell
pscoast -R-170/-140/50/70 -JM4i -Baf -A0+l -W -Di -Vd > test.ps
```

**Full error message**

<details>
<summary>Click to show...</summary>

```
pscoast: GMT->session.HOMEDIR = /Users/liamtoney
pscoast: GMT->session.USERDIR = /Users/liamtoney/.gmt [created]
pscoast: GMT->session.CACHEDIR = /Users/liamtoney/.gmt/cache [created]
pscoast: GMT: 1. gmt_getsharepath trying current dir
pscoast: GMT: 2. gmt_getsharepath trying USERDIR /Users/liamtoney/.gmt
pscoast: GMT: 3. gmt_getsharepath trying USERDIR subdir /Users/liamtoney/.gmt/postscriptlight
pscoast: GMT: 4. gmt_getsharepath trying SHAREDIR subdir /usr/local/Cellar/gmt/5.4.4_2/share/gmt/postscriptlight
pscoast: Map distance calculation will be Cartesian
pscoast: The PROJ_GEODESIC set to Vincenty
pscoast: Initialize FFTW with 4 threads.
pscoast: GMT_Create_Session initialized GMT structure
pscoast: Shared Library # 0 (core). Path = libgmt.dylib
pscoast: Loading core GMT shared library: libgmt.dylib
pscoast: Loading GMT plugins from: /usr/local/Cellar/gmt/5.4.4_2/lib/gmt/plugins
pscoast: Shared Library # 1 (supplements). Path = /usr/local/Cellar/gmt/5.4.4_2/lib/gmt/plugins/supplements.so
pscoast: History: Process -R-170/-140/50/70.
pscoast: History: Process -JM4i.
pscoast: History: Process -Baf.
pscoast: Map distance calculation will be using great circle approximation with authalic auxiliary latitudes and authalic (R_2) radius = 6371007.1809 m, in meter.
pscoast: Central meridian not given, default to -155
pscoast: Projected values in meters: -1.66979e+06 1.66979e+06 6.41352e+06 1.10285e+07
pscoast: Auto-frame interval for axis 0 item 0: d = 10  f = 2
pscoast: Auto-frame interval for axis 1 item 0: d = 5  f = 1
pscoast: Map scale is 328.699 km per cm or 1:3.28699e+07.
pscoast: 1. GSHHG: GSHHGDIR set, trying /tmp/gmt-20181019-68987-1qj1qvm/gmt-5.4.4/gshhg/binned_GSHHS_i.nc
pscoast: 1. GSHHG: Failure, could not access /tmp/gmt-20181019-68987-1qj1qvm/gmt-5.4.4/gshhg/binned_GSHHS_i.nc
pscoast: GMT: 1. gmt_getsharepath trying current dir
pscoast: GMT: 2. gmt_getsharepath trying USERDIR /Users/liamtoney/.gmt
pscoast: GMT: 3. gmt_getsharepath trying USERDIR subdir /Users/liamtoney/.gmt/conf
pscoast: GMT: 4. gmt_getsharepath trying SHAREDIR subdir /usr/local/Cellar/gmt/5.4.4_2/share/gmt/conf
pscoast: GMT: 5. gmt_getsharepath failed
pscoast: GMT: 1. gmt_getsharepath trying current dir
pscoast: GMT: 2. gmt_getsharepath trying USERDIR /Users/liamtoney/.gmt
pscoast: GMT: 3. gmt_getsharepath trying USERDIR subdir /Users/liamtoney/.gmt/coast
pscoast: GMT: 4. gmt_getsharepath trying SHAREDIR subdir /usr/local/Cellar/gmt/5.4.4_2/share/gmt/coast
pscoast: GMT: 5. gmt_getsharepath failed
pscoast: 3. GSHHG: Trying via sharepath
pscoast: GMT: 1. gmt_getsharepath trying current dir
pscoast: GMT: 2. gmt_getsharepath trying USERDIR /Users/liamtoney/.gmt
pscoast: GMT: 3. gmt_getsharepath trying USERDIR subdir /Users/liamtoney/.gmt/coast
pscoast: GMT: 4. gmt_getsharepath trying SHAREDIR subdir /usr/local/Cellar/gmt/5.4.4_2/share/gmt/coast
pscoast: 3. GSHHG: Trying /usr/local/Cellar/gmt/5.4.4_2/share/gmt/coast/binned_GSHHS_i.nc
pscoast: 3. GSHHG: OK, could access /usr/local/Cellar/gmt/5.4.4_2/share/gmt/coast/binned_GSHHS_i.nc
pscoast: NetCDF Library Version: 4.6.1 of Oct 19 2018 20:25:27 $
pscoast: GSHHG with two Antarcticas, read in extra ANT flgs.
pscoast: GSHHG version 2.3.7
Derived from World Vector Shoreline, CIA WDB-II, and Atlas of the Cryosphere
Processed by Paul Wessel and Walter H. F. Smith, 1994-2017
ERROR: Caught signal number 11 (Segmentation fault) at
0   libgmt.5.dylib                      0x000000010aeb4beb gmt_get_shore_bin + 1642
1   ???                                 0x0000000000000774 0x0 + 1908
Stack backtrace:
0   libgmt.5.dylib                      0x000000010ae4d37d sig_handler + 561
1   libsystem_platform.dylib            0x00007fff7f1ddb3d _sigtramp + 29
2   ???                                 0x0000000000000000 0x0 + 0
3   libgmt.5.dylib                      0x000000010b0584b3 GMT_pscoast + 11131
4   libgmt.5.dylib                      0x000000010ae5ac55 GMT_Call_Module + 629
5   pscoast                             0x000000010ae45177 main + 1281
6   libdyld.dylib                       0x00007fff7eff408d start + 1
```
</details>
</br>

**System information**

* Operating system: macOS Mojave 10.14.1
* Version of GMT: 5.4.4 (r20314) [64-bit] [4 cores]

**Comment by joa-quim** [comment]
Works fine fine for me with master but I don't so many warning messages trying to find the coastlines database.

**Comment by seisman** [comment]
I have the same issue on macOS Mojave 10.14.2 with latest master (6.0.0_cdaecae).

**Comment by PaulWessel** [comment]
Workaround is to give something larger than 0 in -A, e.g. -A0.000001+l.  I have fixed the bug and will do a pull request soon.

**Comment by seisman** [comment]
Fixed in #232. Closed.
